src/protocols/displayport.stanza

```diff
@@ -0,0 +1,220 @@
+
+This function will construct the lane connections of a DisplayPort connection. The `src` and `dst`
+arguments must both be of type `displayport`.
+
+@param sw The intra-pair skew time difference for the differential signals being connected.
+@param ml The maximum loss in dB for all differential signals being connected.
+@param cap Optional DC blocking capacitor to be inserted between the TX and RX pairs. Note that
+there are 2 blocking caps inserted, one on each of the P,N signals of the pair.
+@param src This is assumed to be the "Tx" side of the link. This is expected to be a port of `Bundle` type `displayport`
+@param dst This is assumed to be the "Rx" side of the link. This is expected to be a port of `Bundle` type `displayport`
+
+<DOC>
+public defn connect-displayport-lane (sw:Toleranced, ml:Double, cap:Instantiable|False, src:JITXObject, dst:JITXObject) :
+  inside pcb-module:
+    for i in indices(src.lane) do :
+      match(cap) :
+        (c:Instantiable) :
+          inst bl-cap : dp-coupler(c)
+          require tx1 : dual-pair from bl-cap
+          topo-net(src.lane[i], tx1.A)
+          topo-net(tx1.B, dst.lane[i])
+        (f:False) :
+          topo-net(src.lane[i], dst.lane[i])
+      constrain-ch(sw, ml, src.lane[i], dst.lane[i])
+
+doc: \<DOC>
+@brief Construct aux connection of a DisplayPort connection.
+
+This function will construct the AUX channel of a DisplayPort connection with the 
+optional blocking cap and pull-up resistors.
+The `src` and `dst` arguments must both be of type `displayport`.
+
+@param sw The intra-pair skew time difference for the differential signals being connected.
+@param ml The maximum loss in dB for all differential signals being connected.
+@param res Optional pull-up resistor to be added to the AUX pair before the capacitor. 
+Note that there are 2 resistors added, one on each of the P,N signals of the pair.
+@param vdd Power supply pin to be used for the optional pull-up resistors. Must be present if the resistors are used.
+@param cap Optional DC blocking capacitor to be inserted between the AUX pair. Note that
+there are 2 blocking caps inserted, one on each of the P,N signals of the pair.
+@param src This is assumed to be the source side of the link. This is expected to be a port of `Bundle` type `displayport`
+@param dst This is assumed to be the sink side of the link. This is expected to be a port of `Bundle` type `displayport`
+
+<DOC>
+public defn connect-displayport-aux (sw:Toleranced, ml:Double, res:Instantiable|False, vdd:JITXObject|False, cap:Instantiable|False, src:JITXObject, dst:JITXObject) :
+  inside pcb-module:
+    match(res, vdd, cap) :
+      (r:Instantiable, v:JITXObject, c:Instantiable) :
+        inst shunt-r : symmetric-shunt(r)
+        inst bl-cap : dp-coupler(c)
+        require tx1 : dual-pair from bl-cap
+        topo-net(src.aux, shunt-r.A)
+        topo-net(shunt-r.B, tx1.A)
+        topo-net(tx1.B, dst.aux)
+        topo-net(shunt-r.COMMON, v)
```

Comment:
Ya - there are a lot of args and connections here. 

This makes me want to do something like [Network](https://github.com/callendorph/etools/blob/master/src/jitx/Network.stanza) - but I don't think we know the abstraction yet.

Reply:
Yep -- food for thought.